overlord/snapstate/snapstate.go

```diff
@@ -340,15 +340,15 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		return installSet, nil
 	}
 
-	var confFlags int
-	if !snapst.IsInstalled() && snapsup.SideInfo != nil && snapsup.SideInfo.SnapID != "" {
-		// installation, run configure using the gadget defaults
-		// if available
-		confFlags |= UseConfigDefaults
-	}
-
 	// we do not support configuration for bases or the "snapd" snap yet
 	if snapsup.Type != snap.TypeBase && snapsup.Type != snap.TypeSnapd {
+		var confFlags int
+		if !snapst.IsInstalled() && snapsup.SideInfo != nil && snapsup.SideInfo.SnapID != "" && snapsup.Type != snap.TypeOS {
+			// installation, run configure using the gadget defaults
+			// if available, system config defaults are triggered
+			// only during seeding
```

Comment:
// if available, system config defaults (attached to "core") are consumed
// only during seeding,  via an explicit configure step separate from installing

Also thinking further because ConfigureSnap will look at instance name, maybe the check should be snasup.InstanceName() != "core"